### PR TITLE
[SPARK-47227][FOLLOW][DOCS] Improve Spark Connect Documentation

### PR DESCRIPTION
**Pull request by grundprinzip**
### What changes were proposed in this pull request?
Language improvements from https://github.com/apache/spark/pull/45335.

### Why are the changes needed?
Readability.

### Does this PR introduce _any_ user-facing change?
Documentation only.

### How was this patch tested?
N/A

### Was this patch authored or co-authored using generative AI tooling?
No

### CODE REVIEW

File: docs/spark-connect-overview.md [hunk 1]
```diff
@@ -56,7 +56,7 @@ client through gRPC as Apache Arrow-encoded row batches.
   <img src="img/spark-connect-communication.png" title="Spark Connect communication" alt="Spark Connect communication" />
 </p>
 
-## What is changing with Spark Connect
+## How Spark Connect client applications differ from classic Spark applications
```
Comment (grundprinzip) [inline docs/spark-connect-overview.md:59]: @nchammas what about this proposal?

Comment (nchammas) [reply docs/spark-connect-overview.md:59]: This is much clearer. 👍

**Review by xinrong-meng — APPROVED** [review]

**Comment by xinrong-meng** [comment]
LGTM, thank you! Looks much clearer.

**Review by MaxGekk — APPROVED** [review]

**Comment by MaxGekk** [comment]
+1, LGTM. Merging to master.
Thank you, @grundprinzip and @xinrong-meng @nchammas for review.

**Pull request closed without merging**